src/nft/strings.cc

```diff
@@ -393,7 +393,6 @@ Nft ReluctantReplace::reluctant_leftmost_nft(nfa::Nfa nfa, Alphabet* alphabet, S
     Nft nft_reluctant_leftmost{
         nft::builder::create_from_nfa(nfa, 2, { EPSILON }, { EPSILON }) };
     const size_t regex_num_of_states{ nft_reluctant_leftmost.num_of_states() };
-    assert(nft_reluctant_leftmost.is_deterministic());
```

Comment:
We need to check whether the algorithm still works correctly when using NFAs here.

Reply:
(Issue created.)